src/main/java/org/assertj/core/api/AbstractStringAssert.java

```diff
@@ -375,4 +376,26 @@ public SELF isEqualTo(String expectedStringTemplate, Object... args) {
   public SELF isEqualTo(String expected) {
     return super.isEqualTo(expected);
   }
+
+  /**
+   * Verifies that the actual value is equal to expected on their canonical form.
+   * <p>
+   * This method normalizes the expected {@link String} so that the comparison works as expected.
+   *
+   * Examples:
+   * <pre><code class='java'> // assertion succeeds
+   * assertThat(&quot;\u00C4&quot;).isEqualToNormalizingUnicode(&quot;\u0041\u0308&quot;);
+   *
+   *  // assertion fails
+   *  assertThat(&quot;\u0041\u0308&quot;).isEqualToNormalizingUnicode(&quot;\u0041\u0308&quot;); </code></pre>
+   *
+   * @param expected the given {@link String} to compare the actual to.
+   * @return this assertion object.
+   * @see Assert#isEqualTo(java.lang.Object)
+   * @since 3.13.0
+   */
+  public SELF isEqualToNormalizingUnicode(String expected) {
+    String expectedNormalized = Normalizer.normalize(expected, Normalizer.Form.NFC);
+    return super.isEqualTo(expectedNormalized);
```

Comment:
In order to have a specific error message explaining that strings were Unicode normalized, we need to follow the same approach as  `assertEqualsNormalizingPunctuationAndWhitespace`, this means:
* add `assertEqualsNormalizingUnicode` to `Strings` and call it here (need unit test for it too)
* add `ShouldBeEqualNormalizingUnicode` error + unit tests

Reply:
this method should actually be in `AbstractCharSequenceAssert`